### PR TITLE
support for versioned yaml reqs

### PR DESCRIPTION
**Pull request by sopos**
This PR implement changes needed by https://github.com/beakerlib/beakerlib/issues/96.

### CODE REVIEW

**Review by hegerj — APPROVED** [review]

**Comment by hegerj** [comment]
I must admit that the code is very hard to read for me :)

**Comment by sopos** [comment]
I will try to add some comments to make it easier

File: src/rpms.sh [hunk 1]
```diff
@@ -465,43 +479,91 @@ Returns number of unsatisfied requirements.
...
+        # and process it
+        __INTERNAL_req_check 'required binary'
       else
-        package="$(rpm -q --whatprovides "$req" 2> /dev/null)"
+        # the requirment was not a command neither, try generic rpm require, get NEVRA
+        NEVRA=( $(rpm -q --qf "%{name} %{epoch} %{version} %{release} %{arch}" --whatprovides "$req" 2> /dev/null) )
         if [[ $? -eq 0 ]]; then
-          LOG=("${LOG[@]}" "$package" "covers requirement '$req'")
-          rljRpmLog "$package"
+          # the require is know by rpm, process it
```
Comment (hegerj) [inline src/rpms.sh:540]: know -> known

**Review by hegerj — CHANGES_REQUESTED** [review]
Thanks for the comments.
Found couple more typos and possibly unused variable.

File: src/rpms.sh [hunk 2]
```diff
@@ -465,43 +479,91 @@ Returns number of unsatisfied requirements.
...
+    # construct a log message without a version test
+    LOG+=("covers $msg '$req'" 0 )
+  fi
+  # log the package information to journal anyway
+  rljRpmLog "$package"
+}
+
 rlCheckRequirements() {
-  local req res=0 package binary provides LOG=() LOG2 l=0 ll
+  local req ver op res=0 package binary provides LOG=() LOG2 l=0 ll NEVRA
```
[the remark this answers is not in the document]
Comment (sopos) [reply src/rpms.sh:518]: apparently not

**Comment by sopos** [comment]
hopefully last check

**Review by hegerj — APPROVED** [review]
Looks good, thank you for implementing this.

**Pull request merged**